pkg/controller/download_controller.go

```diff
@@ -253,7 +255,19 @@ func (c *downloadController) processDownload(req *pluginv1api.Download) error {
 		}
 	}
 	log.Infof("Copy options: %v, source PEID: %s, target PEID: %s", options, peID.String(), targetPEID.String())
-	returnPeId, err := c.dataMover.CopyFromRepo(peID, targetPEID, options)
+	var returnPeId astrolabe.ProtectedEntityID
+	if req.Spec.BackupRepositoryName != "" {
```

Comment:
Shouldn't you also check backupRepositoryName != constants.WithoutBackupRepository here in addition to ""?